### PR TITLE
chore: reduce subscription boilerplate with `ClrDestroyService`

### PR DESCRIPTION
**Pull request by arturovt**

### CODE REVIEW

File: projects/angular/src/popover/dropdown/dropdown.ts
Comment (mathisscott) [inline projects/angular/src/popover/dropdown/dropdown.ts:45]: same question regarding `changes` here...

**Review by mathisscott — CHANGES_REQUESTED** [review]

**Review by mathisscott — APPROVED** [review]

File: projects/angular/src/accordion/stepper/stepper.ts
Comment (Jinnie) [inline projects/angular/src/accordion/stepper/stepper.ts:45]: Why do we switch to using `@Inject` here?

Comment (arturovt) [reply projects/angular/src/accordion/stepper/stepper.ts:45]: This is used to explicitly add the `| null` union, since the `formGroup` _might be null_, because of the `@Optional()` flag.

File: projects/angular/src/data/datagrid/render/main-renderer.ts
Comment (Jinnie) [inline projects/angular/src/data/datagrid/render/main-renderer.ts:84]: Why making the headers and rows public?

File: projects/angular/src/layout/tabs/tabs.ts
Comment (Jinnie) [inline projects/angular/src/layout/tabs/tabs.ts:106]: why not private again?

File: projects/angular/src/data/datagrid/render/cell-renderer.ts
Comment (Jinnie) [inline projects/angular/src/data/datagrid/render/cell-renderer.ts:42]: Nice! :)

**Review by Jinnie — COMMENTED** [review]
Thanks for this contribution!
I finally finished reviewing it (sorry for the delay). After few minor questions, I'm okay to approve and merge the changes.
The cleanup pattern is rather interesting, I like it and would use it in future too, though TBH, I wouldn't prioritize myself refactoring it all over the codebase at this stage.
Great thanks for all this effort!

**Review by mathisscott — CHANGES_REQUESTED** [review]
Putting on hold for a bit so the team can discuss.

File: projects/angular/src/data/datagrid/render/main-renderer.ts
Comment (arturovt) [reply projects/angular/src/data/datagrid/render/main-renderer.ts:84]: Reverted.

File: projects/angular/src/layout/tabs/tabs.ts
Comment (arturovt) [reply projects/angular/src/layout/tabs/tabs.ts]: Reverted.

File: projects/angular/src/accordion/stepper/stepper.ts
Comment (kevinbuhmann) [reply projects/angular/src/accordion/stepper/stepper.ts:45]: I feel like adding `| null` annotations is entirely unrelated and shouldn't be part of this PR/commit.

**Comment by Jinnie** [comment]
I respect all work done, but due to some internal processes and after discussion with the team, we prefer to refrain from merging this specific PR at this moment. It introduces support risks which, though minor, are not aligned with our current focus areas and we can't afford in this moment.
We may get back to it in future iterations, but for now I'm closing it.

**Pull request closed without merging**
